### PR TITLE
Try less and more when pager is not found (on macOS?)

### PR DESCRIPTION
**Pull request by wdoekes**
Closes: #711
Reported-by: Ryan Carsten Schmidt

### CODE REVIEW

File: src/sipp.cpp [hunk 1]
```diff
@@ -767,17 +775,33 @@ static pid_t begin_pager() {
...
-    /* Get pager first, so we can bail if it's not there */
-    argv[0] = getenv("PAGER");
-    if (!argv[0]) {
-        argv[0] = pager; /* missing PAGER */
-    } else if (!*argv[0]) {
-        return 0; /* blank PAGER */
+    /* Get pager first, so we can bail if it's not there. Prefer env */
+    if ((ptr = getenv("PAGER"))) {
+        if (!ptr[0]) {
+            return 0; /* blank PAGER */
```
Comment (orgads) [inline src/sipp.cpp:781]: !ptr

File: src/sipp.cpp [hunk 2]
```diff
@@ -756,7 +756,15 @@ static char* wrap(const char* in, int offset, int size)
 /* If stdout is a TTY, wrap stdout in a call to PAGER (generally less(1)).
  * Returns a pid_t you'll have to pass to end_pager(). */
 static pid_t begin_pager() {
-    char pager[15] = "/usr/bin/pager";
+    char const *pager_options[4] = {
```
Comment (orgads) [inline src/sipp.cpp:759]: `std::vector<std::string>`? Iteration is easier, and you won't need a copy, you can pass `&opt[0]`.

**Review by orgads — CHANGES_REQUESTED** [review]

File: src/sipp.cpp (hunk 1 above)
Comment (wdoekes) [reply src/sipp.cpp:781]: No. `!ptr[0]` or `ptr[0] == '\0'`

The `if` at 779 already checks `!ptr`.

File: src/sipp.cpp (hunk 2 above)
Comment (wdoekes) [reply src/sipp.cpp:759]: :shrug: 

I guess we should do more c++. Be my guest :grimacing: 

I assume we won't need a copy because it _is_ a copy already.

**Review by orgads — APPROVED** [review]
You can merge this as it is, I'll C++ify it later :)

**Comment by wdoekes** [comment]
closed by 3242112e3fb0d003b15d8ddaade4d468f01068f1

**Pull request closed without merging**